internal/backend/usagereporter/usagedata.go

```diff
@@ -0,0 +1,145 @@
+package usagereporter
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/google/uuid"
+	"go.autokitteh.dev/autokitteh/internal/version"
+	"go.autokitteh.dev/autokitteh/internal/xdg"
+	"go.uber.org/zap"
+)
+
+type reportRequest struct {
+	InstallationID string `json:"installation_id"`
+	Version        string `json:"version"`
+	Commit         string `json:"commit"`
+	OS             string `json:"os"`
+	Arch           string `json:"arch"`
+}
+
+type UsageReporter interface {
+	Start()
+	Stop()
+}
+
+type usageReporter struct {
+	installationID uuid.UUID
+	updateInterval time.Duration
+	shutdownChan   chan struct{}
+	endpoint       string
+	logger         *zap.Logger
+}
+
+func generateInstallIDFile(path string) error {
+	id := uuid.New()
+	if err := os.WriteFile(path, []byte(id.String()), 0600); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ensureAndReadInstallationIDFile(installationIDFile string) (string, error) {
+	// Try read installationID file,
+	// if there is an error and the file exists, try to remove it and fail if cant
+	// generate a new file
+	// try read the data again
+	data, err := os.ReadFile(installationIDFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			if err := os.Remove(installationIDFile); err != nil {
+				return "", err
+			}
+		}
```

Comment:
Why is this needed? According to https://pkg.go.dev/os#WriteFile WriteFile() rewrites the file if it exists, no?

Reply:
good point, removed